hcmanager/classification_filter.go

```diff
@@ -0,0 +1,167 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hcmanager
+
+import (
+	hwcc "hardware-classification-controller/api/v1alpha1"
+
+	bmh "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+)
+
+// MinMaxFilter it will perform the minimum and maximum comparison based on the value provided by the user and check for the valid host
+func (mgr HardwareClassificationManager) MinMaxFilter(ProfileName string, HostList []bmh.HardwareDetails, expectedHardwareprofile hwcc.HardwareCharacteristics) []string {
+	var validHosts []string
+	for _, hardwareDetail := range HostList {
+		if !checkCPUCount(mgr, hardwareDetail.CPU, expectedHardwareprofile.Cpu, hardwareDetail.Hostname) ||
+			!checkRAM(mgr, hardwareDetail.RAMMebibytes, expectedHardwareprofile.Ram, hardwareDetail.Hostname) ||
+			!checkNICS(mgr, len(hardwareDetail.NIC), expectedHardwareprofile.Nic, hardwareDetail.Hostname) ||
+			!checkDiskDetails(mgr, hardwareDetail.Storage, expectedHardwareprofile.Disk, hardwareDetail.Hostname) {
+			continue
+		}
+		validHosts = append(validHosts, hardwareDetail.Hostname)
+	}
+	return validHosts
+}
+
+//checkCPUCount this function checks the CPU details for both min and max parameters
+func checkCPUCount(mgr HardwareClassificationManager, cpu bmh.CPU, expectedCPU *hwcc.Cpu, bmhName string) bool {
+	if expectedCPU == nil {
+		return true
+	}
+	if expectedCPU.MaximumCount > 0 {
+		expectedMaxCPUCount := expectedCPU.MaximumCount
+		mgr.Log.Info("Maximum CPU Count", "Expected", expectedMaxCPUCount, "Actual", cpu.Count)
+		if expectedMaxCPUCount < cpu.Count {
+			mgr.Log.Info("CPU Count Mismatched", "BareMetalHost", bmhName)
+			return false
+		}
+	}
+	if expectedCPU.MinimumCount > 0 {
+		expectedMinCPUCount := expectedCPU.MinimumCount
+		mgr.Log.Info("Minimum CPU Count", "Expected", expectedMinCPUCount, "Actual", cpu.Count)
+		if expectedMinCPUCount > cpu.Count {
+			mgr.Log.Info("CPU Count Mismatched", "BareMetalHost", bmhName)
+			return false
+		}
+	}
```

Comment:
you could make a util function out of that to avoid repetition, taking as parameter the element to check, the min and the max, and the name for the log, something like : 
```
func compare(item, min, max int, name, bmh string) bool {
    if min > 0 && item < min {
        mgr.Log.Info(fmt.Sprintf("%v Mismatched, expected %v to be higher than %v", name, item, min), "BMH", bmh)
        return false
    }
    if max > 0 && item > max {
        mgr.Log.Info(fmt.Sprintf("%v Mismatched, expected %v to be lower than %v", name, item, max), "BMH", bmh)
        return false
    }
}
```

Reply:
you might need type conversion to use it, and maybe with floats rather than ints depending on what you are comparing

Reply:
I am thinking to address all these comments in next PR. Also I would like to keep mod file like other Metal3 projects with same path. 
Anyway we will need another PR to make it more optimize and standardize it aligned with CAPM3 and BMO projects.